llvm/test/Feature/Repo/repo_pruning_dependents.ll

```diff
@@ -1,27 +1,47 @@
-; This testcase tests the program repository pruning pass.
+; This testcase tests the program repository pruning pass for the dependents.
 ;
-; If function 'A' is dependent on function 'B' and 'B' is dependent
-; on function 'C', both TickeNodes of 'B' and 'C' need to be added
-; into in the 'repo.tickets' during the pruning.
+; In the llvm-project-prepo project, the GO's digest could be generated either by the
+; RepoHashGenerator pass or by the RepoObjectWriter. When a new global object (GO) is
+; generated by an LLVM optimisation pass after the RepoHashGenerator pass, the digest
+; will be generated by the RepoObjectWriter. If a fragment has an external fixup
+; referencing this GO, we add the GO's compilation member address to the dependents.
+;
+; When a fragment is pruned, all its dependents need to be recorded in the repo.tickets
+; metadata to guarantee that the dependents (those GO generated by later optimizations)
+; are present in the final compilation record.
+;
+; In this test, the digest of 'A' is given in the TicketNode. The digests of `B` and `C`
```

Comment:
You could say in the first sentence:

"In this test, the digest of 'A' is given in the TicketNode (as if it were generated by the RepoHashGenerator pass)."

That would help tie it up with the earlier paragraph.

Reply:
Good point, added and thanks.